client/controllers/search.controller.js

```diff
@@ -0,0 +1,29 @@
+(function() {
+    "use strict";
+    
+    angular
+        .module("search", [])
+        .controller("SearchController", SearchController);
+    
+    SearchController.$inject = ["$scope", "$http", "regexEscapeService"];
+    
+    function SearchController($scope, $http, regexEscapeService) {
+        var vm = this;
+        
+        vm.searchFormString = "";
+
+        vm.search = function() {
+            var req = {
+                payload: regexEscapeService.escapeRegex(vm.searchFormString)
+            };
+
+            $http.post("/search", req)
+                .then(function(response) {
+                    vm.words = response.data;
+                },
+                function(response) {
+                    console.log(response.data);
+                });
+        };
+    }
+})();
```

Comment:
I made lots of changes here. In summary, I removed `$scope.$watch` in favor of `ng-change`. This way we don't muddy up scope and it cleans up the code. I also merged `getWords()` and the `$watch` logic into a new `search()` function since we're no longer getting the list until we search.

I also refactored `$scope.search` into `vm.searchFormString` since we no longer use `$watch` and it prevents muddying scope.